edx_proctoring/templates/proctored_exam/ready_to_start.html

```diff
@@ -56,16 +56,33 @@ <h3>
 {% include 'proctored_exam/footer.html' %}
 
 <script type="text/javascript">
+  var edx = edx || {};
+  edx.courseware = edx.courseware || {};
+  edx.courseware.proctored_exam = edx.courseware.proctored_exam || {};
+  edx.courseware.proctored_exam.configuredWorkerURL = "{{ backend_js_bundle }}";
+
 
   $('.proctored-enter-exam').click(
     function(e) {
       e.preventDefault();
       e.stopPropagation();
 
-      var action_url = $(this).data('change-state-url');
-      var exam_id = $(this).data('exam-id');
-      var action = $(this).data('action');
+        var action_url = $(this).data('change-state-url');
+        var exam_id = $(this).data('exam-id');
+        var action = $(this).data('action');
 
+        var proctoringBackendWorker = new Worker(edx.courseware.proctored_exam.configuredWorkerURL);
+        new Promise((resolve) => {
+            var responseHandler = (e) => {
+                if (e.data.type === "examAttemptStarted") {
+                    proctoringBackendWorker.removeEventListener('message', responseHandler);
+                    proctoringBackendWorker.terminate();
+                    resolve();
+                }
+            };
+            proctoringBackendWorker.addEventListener('message', responseHandler);
```

Comment:
`static/index.js` currently only contains code that gets bundled into the webworker bundle; there isn't a simple way to get code from there into the webpage's main JS `window` context as of right now.

The more available alternatives are to add another file into https://github.com/edx/edx-platform/blob/6c5ee7209c9c0d62b143a93dc43475806c44ec90/lms/envs/common.py#L1351, which would put our JS on every single LMS page whenever proctoring is enabled. That'd be pretty glaringly not ideal, so I'm being glaring in my hackaround.

Reply:
What about ``proctoring/js/proctored_app.js``? That will be on every proctored page, but not every LMS page, right?

Reply:
Nope, every single file in that list is loaded on every page of a course with proctoring enabled: https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/course_navigation.html#L26

Fixing that situation seemed like it'd inflate scope, but we can talk through it synchronously.

Reply:
There is a footer.html template included on every proctored page. It already has javascript and could be used for these event handlers. Ultimately, I think the js is going to have to be refactored into the static/index.js or another file, because once the proctored exam starts, we're in the regular courseware context, without any templates available from edx-proctoring. We won't be able to create the event listener for the application heartbeat, unless the code is refactored.

Reply:
referenced in https://openedx.atlassian.net/browse/EDUCATOR-3690